crates/solver/src/settlement_submission.rs

```diff
@@ -82,23 +80,25 @@ impl SolutionSubmitter {
             TransactionStrategy::ArcherNetwork {
                 archer_api,
                 max_confirm_time,
+                retry_interval,
             } => {
-                let submitter = ArcherSolutionSubmitter::new(
+                let submitter = Submitter::new(
                     &self.web3,
                     &self.contract,
                     &account,
                     archer_api,
                     self.gas_price_estimator.as_ref(),
-                    self.gas_price_cap,
                 )?;
-                let result = submitter
-                    .submit(
-                        self.target_confirm_time,
-                        SystemTime::now() + *max_confirm_time,
-                        settlement,
-                        gas_estimate,
-                    )
-                    .await;
+                let params = SubmitterParams {
+                    target_confirm_time: self.target_confirm_time,
+                    gas_estimate,
+                    gas_price_cap: self.gas_price_cap,
```

Comment:
Why couldn't this be part of the `Submitter` and instead is in the `SubmitterParams`?

Reply:
It could be, however I aimed here to decouple the parameters that influence the execution path of the `Submitter`. This way, `Submitter` will be the same for all strategies, while `SubmitterParams` will be different for each strategy.
Also, not completely sure what will be our needs in the next PR when we implement point 4 from https://github.com/gnosis/gp-v2-services/issues/1407